rdrf/rdrf/testing/behaviour/features/steps.py

```diff
@@ -844,22 +844,43 @@ def scroll_to_centre(xp):
 
 @step('I add patient name "(.*)" sex "(.*)" birthdate "(.*)"')
 def add_new_patient(step, name, sex, birthdate):
-    # Woud like to make this a bit more smooth in future,
-    # to avoid repeated calls to find.
+    # Refactored 20200414 to avoid writing "find" multiple times in succession.
+    # Prepare name, page, and lists;
     surname, firstname = name.split()
     world.browser.get(
         world.site_url + "patientslisting"
     )
-    find("//button[@id='add_patient']").click()
-    find("//option[contains(., 'ICHOM Colorectal Cancer')]").click()
-    find("//option[contains(., 'ICHOMCRC SJOG')]").click()
-    find("//input[@name='family_name']").send_keys(surname)
-    find("//input[@name='given_names']").send_keys(firstname)
-    find("//input[@name='date_of_birth']").send_keys(birthdate, Keys.ESCAPE)
-    scroll_to_centre("//select[@name='sex']")
-    find("//select[@name='sex']").click()
-    find("//select[@name='sex']/option[text()='%s']" % sex).click()
-    find("//button[@id='submit-btn']").click()
+    xp_list = [
+        "//button[@id='add_patient']",
+        "//option[contains(., 'ICHOM Colorectal Cancer')]",
+        "//option[contains(., 'ICHOMCRC SJOG')]",
+        "//input[@name='family_name']",
+        "//input[@name='given_names']",
+        "//input[@name='date_of_birth']",
+        "//select[@name='sex']",
+        "//select[@name='sex']/option[text()='%s']" % sex,
```

Comment:
Better to name these like:
class Xpath:
   ADD_PATIENT_BUTTON = "//button[@id='add_patient']"

then yon can refer later

like Xpath.ADD_PATIENT_BUTTON and you immediately know you're trying to do

Not sure I like you putting into xp_list here